backend/tests/test_course_build_processor.py

```diff
@@ -0,0 +1,478 @@
+    user_id = setup_new_user_manager.id
+    course_build = CourseBuild(
+        instructor_firstname="Illidan",
+        instructor_lastname="Stormrage",
+        instructor_email="istormrage@blacktemple.edu",
+        course_name="Algebra 1 - Illidan Stormrage (AY 2024)",
+        course_shortname="Alg1 IS AY24",
+        course_category=6,
+        course_id=None,
+        course_enrollment_url=None,
+        course_enrollment_key=None,
+        school_district_id=school_district_id,
+        academic_year="AY 2024",
+        academic_year_short="AY24",
+        base_course_id=70,
+        status="created",
+        creator_id=user_id,
+    )
+    course_build2 = CourseBuild(
+        instructor_firstname="Thomas",
+        instructor_lastname="Stormrage",
+        instructor_email="tstormrage@blacktemple.edu",
+        course_name="Algebra 1 - Thomas Stormrage (AY 2024)",
+        course_shortname="Alg1 TS AY24",
+        course_category=6,
+        course_id=None,
+        course_enrollment_url=None,
+        course_enrollment_key=None,
+        school_district_id=school_district_id,
+        academic_year="AY 2024",
+        academic_year_short="AY24",
+        base_course_id=70,
+        status="processing",
+        creator_id=user_id,
+    )
+    course_build3 = CourseBuild(
+        instructor_firstname="Michael",
+        instructor_lastname="Stormrage",
+        instructor_email="mstormrage@blacktemple.edu",
+        course_name="Algebra 1 - Michael Stormrage (AY 2024)",
+        course_shortname="Alg1 MS AY24",
+        course_category=6,
+        course_id=None,
+        course_enrollment_url=None,
+        course_enrollment_key=None,
+        school_district_id=school_district_id,
+        academic_year="AY 2024",
+        academic_year_short="AY24",
+        base_course_id=70,
+        status="completed",
+        creator_id=user_id,
+    )
+    db.add(course_build)
+    db.add(course_build2)
+    db.add(course_build3)
+    db.commit()
+
+
+def test_course_build_processor(mocker, db, create_course_builds):
+    sqs_client = boto3.client("sqs", region_name="azeroth")
+    sqs_stubber = botocore.stub.Stubber(sqs_client)
+
+    course_builds = db.query(CourseBuild).all()
+
+    initial_course_build = course_builds[0]
+
+    mock_sqs_data = {"course_build_id": initial_course_build.id}
+
+    mock_settings = mocker.Mock()
+    setattr(mock_settings, "SQS_QUEUE", "testqueue")
+    setattr(mock_settings, "SQS_POLL_INTERVAL_MINS", "1")
+    mocker.patch(
+        "rope.api.processors.course_build_processor.settings",
+        mock_settings,
+    )
+
+    engine = create_engine("postgresql://pguser:pgpassword@localhost/ropedb")
+    mock_factory = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+    mocker.patch(
+        "rope.api.processors.course_build_processor.get_db",
+        lambda: mock_factory,
+    )
+
+    mocker.patch(
+        "rope.api.processors.course_build_processor.get_moodle_user_role_by_shortname",
```

Comment:
Instead of mocking out this function, it might be nicer to mock out `rope.api.routers.moodle.moodle_client.get_role_by_shortname` instead.